oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/TableHelper.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.skywalking.oap.server.core.analysis.DownSampling;
+import org.apache.skywalking.oap.server.core.analysis.FunctionCategory;
+import org.apache.skywalking.oap.server.core.analysis.TimeBucket;
+import org.apache.skywalking.oap.server.core.config.ConfigService;
+import org.apache.skywalking.oap.server.core.storage.model.Model;
+import org.apache.skywalking.oap.server.library.client.jdbc.hikaricp.JDBCClient;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+import org.apache.skywalking.oap.server.storage.plugin.jdbc.TableMetaInfo;
+
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+/**
+ * Utility class to get table name for a given model.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class TableHelper {
+    private final ModuleManager moduleManager;
+    private final JDBCClient jdbcClient;
+
+    public static String getTableName(Model model) {
+        final var aggFuncName = FunctionCategory.uniqueFunctionName(model.getStreamClass()).replaceAll("-", "_");
+        return StringUtil.isNotBlank(aggFuncName) ? aggFuncName : model.getName();
+    }
+
+    public static String getTableForWrite(Model model) {
+        final var tableName = getTableName(model);
+
+        if (!model.isTimeSeries()) {
+            return tableName;
+        }
+
+        final var dayTimeBucket = TimeBucket.getTimeBucket(System.currentTimeMillis(), DownSampling.Day);
+        return tableName + Const.UNDERSCORE + dayTimeBucket;
+    }
+
+    public static String getTable(Model model, long timeBucket) {
+        final var tableName = getTableName(model);
+        if (timeBucket == 0) {
+            timeBucket = TimeBucket.getTimeBucket(System.currentTimeMillis(), DownSampling.Day);
+        }
+
+        if (!model.isTimeSeries()) {
+            return tableName;
+        }
+
+        return tableName + Const.UNDERSCORE + TimeBucket.getTimeBucket(TimeBucket.getTimestamp(timeBucket), DownSampling.Day);
+    }
+
+    public List<String> getTablesForRead(String modelName, long timeBucketStart, long timeBucketEnd) {
+        final var model = TableMetaInfo.get(modelName);
+        final var tableName = getTableName(model);
+
+        if (!model.isTimeSeries()) {
+            return Collections.singletonList(tableName);
+        }
+
+        timeBucketStart = TimeBucket.getTimeBucket(TimeBucket.getTimestamp(timeBucketStart), DownSampling.Day);
+        timeBucketEnd = TimeBucket.getTimeBucket(TimeBucket.getTimestamp(timeBucketEnd), DownSampling.Day);
+
+        return LongStream
+            .rangeClosed(timeBucketStart, timeBucketEnd)
+            .distinct()
+            .mapToObj(it -> tableName + "_" + it)
+            .filter(table -> {
+                try {
+                    return jdbcClient.tableExists(table);
```

Comment:
I think failing to read from Database in sync may be not worth it. 
1. It could be slow
2. If this kind of query keeps happening, we cost a lot of resources in the database. 

Instead, moving the table cache from `DeleteDao` to JDBC global scope should help. After all, it only read once period.

Reply:
OK, I noticed this is another misguide by naming. This method is from TTL, so, most likely we should rename to `getExistingTables` or similar.

Reply:
> I think failing to read from Database in sync may be not worth it.
> 
> 1. It could be slow
> 2. If this kind of query keeps happening, we cost a lot of resources in the database.
> 
> Instead, moving the table cache from `DeleteDao` to JDBC global scope should help. After all, it only read once period.

Caching these brings other problems, when the tables are deleted in another OAP node, we don't get notified, and still consider this table exists, then we might send queries to an inexistent table, which causes errors.

Reply:
I think we should control the query start time and end time within TTL rather than to query table if exist.

Reply:
It is worth to a cache without loader mechanism to deal with this. 
The metadata reading is low performance. We should not include it into query process.

Reply:
A cache with timeout load mechanism.